projects/maker/index.js

```diff
@@ -36,18 +36,31 @@ async function getJoins(block) {
 
   ilks = ilks.filter((ilk) => ilk.output);
 
-  await Promise.all(ilks.map(async (ilk) => {
+  for (let i = 0; i < ilks.length; i++) {
+    let ilk = ilks[i];
     try {
       let gem = (await sdk.api.abi.call({
         block,
         target: ilk.input.target,
         abi: MakerMCDConstants.gem
       })).output;
+      if (!gem) {
+        // retry in case request fail
+        gem = (await sdk.api.abi.call({
+          block,
+          target: ilk.input.target,
+          abi: MakerMCDConstants.gem
+        })).output;
+      }
+      if (!gem) {
+        throw('no gem')
```

Comment:
Nit: throwing a string vs new Error('no gem') can possibly cause problems when the error is caught yet lacks a `.message` prop

Reply:
ahh good to know, will update that